database/src/main/kotlin/com/eros/database/DatabasePlugin.kt

```
@@ -0,0 +1,92 @@
+
+/**
+ * Ktor plugin that integrates database connectivity, Exposed ORM, and Flyway migrations
+ * into the application lifecycle.
+ *
+ * Initializes on application start and closes connections on application stop.
+ * Runs Flyway migrations automatically to ensure schema is up-to-date.
+ */
+class DatabasePlugin(private val config: DatabaseConfig) {
+
+    lateinit var dataSource: DataSource
+        private set
```

detekt (1.23.8)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)